versions/next/asyncapi.md

```diff
@@ -304,6 +279,7 @@ Field Name | Type | Description
 <a name="serverObjectSchemeVersion"></a>schemeVersion | `string` | The version of the scheme. For instance: AMQP `0.9.1`, Kafka `1.0.0`, etc.
 <a name="serverObjectDescription"></a>description | `string` | An optional string describing the host designated by the URL. [CommonMark syntax](http://spec.commonmark.org/) MAY be used for rich text representation.
 <a name="serverObjectVariables"></a>variables | Map[`string`, [Server Variable Object](#serverVariableObject)] | A map between a variable name and its value.  The value is used for substitution in the server's URL template.
+<a name="serverObjectBaseTopic"></a>baseTopic | `string` | A string describing the base topic. MUST be in the form of a [RFC 3986 URI path](https://tools.ietf.org/html/rfc3986#section-3.3). URI templates are not allowed here. Defaults to empty string.
```

Comment:
I wonder what purpose a specific `baseTopic` property has now? Can't the use either provide a base-topic by simply appending a constant value to the `url` property, or provide a templated variable with the usual `default` and `enum` properties?

Reply:
Unlike HTTP (or OpenAPI basePath), baseTopic is not part of the URL. You can't append the baseTopic to the server URL because it will never connect to the server. However, I think it's a good idea to have it inside the server object because different brokers might have different prepended base topics.

Example:
```
URL: mqtt://broker.company.com
Subscribe: user/created
```

is not the same as

```
URL: mqtt://broker.company.com/user
Subscribe: created
```

It will complain there's no MQTT server listening at `mqtt://broker.company.com/user`.

Reply:
OK, understood.

Reply:
Should we say explicitly that the `url` property cannot contain a `path` or `fragment-id`, or maybe more strictly that it may only contain a `scheme` and a `host` if that is the case?

Reply:
Sorry, my previous example was not clear enough. It doesn't mean you can't have paths or fragment identifiers. A broker URL might look like `amqp://broker.company.com/vhost/1?token=mysecrettoken`, and it's totally valid. Check out [AMQP URI scheme](https://www.rabbitmq.com/uri-spec.html) (scroll down to `2. The "amqp" URI scheme`).

And that's not just AMQP specific. You can stablish a WebSockets connection in a URI similar to the one above. All protocols allow you to pass a fully qualified URI. Not to mention that the server URI might just be the URI of a proxy in the middle.

To your point, fragment identifiers might not be valid since, as per my understanding, they're just for client-side handling, mime-type dependent, and should not produce any impact server-side.